homeassistant/components/victron_ble/sensor.py

```diff
@@ -0,0 +1,99 @@
+"""Sensor platform for Victron BLE."""
+
+import logging
+
+from sensor_state_data import DeviceKey
+
+from homeassistant import config_entries
+from homeassistant.components.bluetooth.passive_update_processor import (
+    PassiveBluetoothDataProcessor,
+    PassiveBluetoothDataUpdate,
+    PassiveBluetoothEntityKey,
+    PassiveBluetoothProcessorCoordinator,
+    PassiveBluetoothProcessorEntity,
+)
+from homeassistant.components.sensor import (
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.sensor import sensor_device_info_to_hass_device_info
+
+from .const import DOMAIN
+from .custom_state_data import SensorDeviceClass
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _victron_state_class(device_key: DeviceKey):
+    """Return the state class for a sensor."""
+    key = device_key.key.lower()
+    if "yield" in key:
+        return SensorStateClass.TOTAL_INCREASING
+    if "alarm" in key or "state" in key or "mode" in key:
+        return None
+    return SensorStateClass.MEASUREMENT
+
+
+def sensor_update_to_bluetooth_data_update(sensor_update):
+    """Convert a sensor update to a bluetooth data update."""
+    return PassiveBluetoothDataUpdate(
+        devices={
+            device_id: sensor_device_info_to_hass_device_info(device_info)
+            for device_id, device_info in sensor_update.devices.items()
+        },
+        entity_descriptions={
+            PassiveBluetoothEntityKey(
+                device_key.key, device_key.device_id
+            ): SensorEntityDescription(
```

Comment:
Define the entity descriptions as constants in a container that we can look up instead. See eg the xiaomi_ble integration.

Reply:
Done. In the new version I'm looking up these constants by `device_key.key` because there are sometimes different keys that have identical device classes and units (for example, "AC Input Power" and "AC Output Power.")